src/main/java/org/folio/rest/impl/InvoiceStorageImpl.java

```diff
@@ -95,13 +99,43 @@ public void postInvoiceStorageInvoices(String lang, Invoice entity, Map<String,
   @Validate
   @Override
   public void getInvoiceStorageInvoicesById(String id, String lang, Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
-    PgUtil.getById(INVOICE_TABLE, Invoice.class, id, okapiHeaders,vertxContext, GetInvoiceStorageInvoicesByIdResponse.class, asyncResultHandler);
+    PgUtil.getById(INVOICE_TABLE, Invoice.class, id, okapiHeaders, vertxContext, GetInvoiceStorageInvoicesByIdResponse.class, asyncResultHandler);
   }
 
   @Validate
   @Override
-  public void deleteInvoiceStorageInvoicesById(String id, String lang, Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
-    PgUtil.deleteById(INVOICE_TABLE, id, okapiHeaders, vertxContext, DeleteInvoiceStorageInvoicesByIdResponse.class, asyncResultHandler);
+  public void deleteInvoiceStorageInvoicesById(String id, String lang, Map<String, String> okapiHeaders,
+      Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+    try {
+      vertxContext.runOnContext(v -> {
+        TxWithId tx = new TxWithId(id);
+        log.info("Delete invoice");
+        startTxWithId(tx).thenCompose(this::deleteInvoiceLinesByInvoiceId)
+          .thenCompose(this::deleteInvoiceById)
```

Comment:
At the moment there is `deleteSequence(entity)` call in `putInvoiceStorageInvoicesById` method however the sequence should be also deleted on invoice deletion. Could you please correct this as part of this change as well?